services/tenant-ui/frontend/src/components/layout/AppLayout.vue

```diff
@@ -30,4 +30,95 @@ import Card from 'primevue/card';
 import Header from './Header.vue';
 import Footer from './Footer.vue';
 import Sidebar from './Sidebar.vue';
+
+// State
+import { storeToRefs } from 'pinia';
+import { useGlobalStateStore } from '@/store/stateStore';
+
+const { sidebarOpen } = storeToRefs(useGlobalStateStore());
+
+const calcOpen = () => {
+  if (sidebarOpen.value === null) {
+    // Use media queries
+    return null;
+  } else if (sidebarOpen.value) {
+    // Default width
+    return 'open';
+  } else {
+    return 'closed'; // Mobile width
+  }
+};
 </script>
+
+<style lang="scss" scoped>
+/* By default the small title is hidden, and the large title is shown */
+:deep(.sidebar-app-title.small) {
+  display: none;
+  font-size: 2rem;
+}
+
+/* When the sidebar is closed */
+@mixin closed {
+  :deep(.p-menuitem-text) {
+    display: none;
+  }
+  :deep(.p-menuitem-icon) {
+    font-size: 2rem !important;
+    margin-left: 0.5rem;
+  }
+  :deep(.sidebar-app-title) {
+    display: none;
+  }
+  :deep(.sidebar-app-title.small) {
+    display: block;
+  }
+}
+
+/* When the sidebar is open */
+@mixin open {
+  :deep(.p-menuitem-text) {
+    display: '';
+  }
+  :deep(.p-menuitem-icon) {
+    font-size: '';
+    margin-left: '';
+  }
+  :deep(.sidebar-app-title) {
+    display: '';
+  }
+  :deep(.sidebar-app-title.small) {
+    display: '';
+  }
+}
+
+/*
+ If the user has not selected a sidebar state, then use media queries to
+  determine the state.
+ */
+.layout-sidebar:not(.open, .closed) {
+  @media (max-width: 1000px) {
+    min-width: 6rem !important;
+    width: 6rem;
+    @include closed;
+  }
+}
+
+/*
+ If the user has selected a sidebar state, then use that state.
+ */
+.layout-sidebar.closed {
+  min-width: 6rem !important;
+  width: 6rem;
+  @include closed;
+}
+.layout-sidebar.open {
+  min-width: '';
+  width: '';
+  @include open;
+}
+
+.layout-sidebar {
+  transition: width 0.2s ease-in-out;
+  transition: min-width 0.2s ease-in-out;
+}
+</style>
```

Comment:
There's a layout.scss file that controls the other stylings from this component. This could go in there.

Reply:
That may be a good spot especially if the Innkeeper is going to look the same.